Travis binaries for OSX are not movable and do not work on non-travis environment

Installing MRI 2.2.5 will pull the binary from travis, but it seems contain hard reference to the system it was built on. Installing _any_ native extension is impossible.

```
rvm -v
rvm 1.27.0 (latest) by Wayne E. Seguin <wayneeseguin@gmail.com>, Michal Papis <mpapis@gmail.com> [https://rvm.io/]

rvm install 2.2.5
Searching for binary rubies, this might take some time.
Found remote file https://rubies.travis-ci.org/osx/10.11/x86_64/ruby-2.2.5.tar.bz2
Checking requirements for osx.
Certificates in '/usr/local/etc/openssl/cert.pem' are already up to date.
Requirements installation successful.
ruby-2.2.5 - #configure
ruby-2.2.5 - #download
ruby-2.2.5 - #validate archive
ruby-2.2.5 - #extract
ruby-2.2.5 - #validate binary
ruby-2.2.5 - #setup
ruby-2.2.5 - #gemset created /Users/USER/.rvm/gems/ruby-2.2.5@global
ruby-2.2.5 - #importing gemset /Users/USER/.rvm/gemsets/global.gems..............................
ruby-2.2.5 - #generating global wrappers........
ruby-2.2.5 - #gemset created /Users/USER/.rvm/gems/ruby-2.2.5
ruby-2.2.5 - #importing gemsetfile /Users/USER/.rvm/gemsets/default.gems evaluated to empty gem list
ruby-2.2.5 - #generating default wrappers........
```

Then:

```
gem install pg

Fetching: pg-0.18.4.gem (100%)
Building native extensions.  This could take a while...
ERROR:  Error installing pg:
    ERROR: Failed to build gem native extension.

# ....

linking shared-object pg_ext.bundle
ld: warning: directory not found for option '-L/Users/travis/.sm/pkg/active/lib'

make "DESTDIR=" install
make: /usr/local/bin/gmkdir: No such file or directory
make: *** [.RUBYARCHDIR.time] Error 1
```

I know it's just a warning, but _ALL_ the installation attempts of gems with native extensions fail with the same error.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -19,6 +19,7 @@
 * Cannot build rbx-2.5.2 on ArchLinux [\#3497](https://github.com/rvm/rvm/issues/3497)
 * Remove incompatible version of openssl098 [\#3844](https://github.com/rvm/rvm/issues/3844) 
 * Failed to fetch the gpg key from keys.gnupg.net [\#3544](https://github.com/rvm/rvm/issues/3544)
+* Filtering Travis binaries for OSX for non Travis env (they are statically linked and not movable) [\#3690](https://github.com/rvm/rvm/issues/3690) 
 
 ## [1.28.0](https://github.com/rvm/rvm/tag/1.28.0)
 

--- a/scripts/functions/utility_system
+++ b/scripts/functions/utility_system
@@ -116,6 +116,13 @@ __list_remote_all()
   while
     __rvm_db "rvm_remote_server_url${_iterator:-}" rvm_remote_server_url
   do
+    if [[ $rvm_remote_server_url == *"travis"* && $TRAVIS != true && $_system_name_lowercase == "osx" ]]
+    then
+        rvm_debug "Travis binaries for OSX are not movable and can't be used outside of Travis environment. \
+Skip that source."
+        continue
+    fi
+
     __rvm_system_path "" "${_iterator}"
     rvm_debug "__list_remote_all${_iterator:-} $rvm_remote_server_url $rvm_remote_server_path"
     __list_remote_for "${rvm_remote_server_url}" "$rvm_remote_server_path"

--- a/scripts/functions/utility_system
+++ b/scripts/functions/utility_system
@@ -61,13 +61,29 @@ __rvm_ruby_package_file()
   esac
 }
 
+__rvm_include_travis_binaries()
+{
+  if [[ $rvm_remote_server_url == *"travis"* && $TRAVIS != true && $_system_name_lowercase == "osx" ]]
+  then
+      rvm_debug "Travis binaries for OSX are not movable and can't be used outside of Travis environment. \
+Skip that source."
+      return 1
+  fi
+
+  return 0
+}
+
 __rvm_calculate_remote_file()
 {
   rvm_remote_server_url="$( __rvm_db "rvm_remote_server_url${3:-}" )"
+
   [[ -n "$rvm_remote_server_url" ]] || {
     rvm_debug "rvm_remote_server_url${3:-} not found"
     return $1
   }
+
+  __rvm_include_travis_binaries || return $1
+
   __rvm_system_path "" "${3:-}"
   __rvm_ruby_package_file "${4:-}"
   __remote_file="${rvm_remote_server_url}/${rvm_remote_server_path}${rvm_ruby_package_file}"
@@ -113,18 +129,15 @@ __list_remote_all()
 {
   \typeset _iterator rvm_remote_server_url rvm_remote_server_path
   _iterator=""
+
   while
     __rvm_db "rvm_remote_server_url${_iterator:-}" rvm_remote_server_url
   do
-    if [[ $rvm_remote_server_url == *"travis"* && $TRAVIS != true && $_system_name_lowercase == "osx" ]]
-    then
-        rvm_debug "Travis binaries for OSX are not movable and can't be used outside of Travis environment. \
-Skip that source."
-        continue
-    fi
+    __rvm_include_travis_binaries || continue
 
     __rvm_system_path "" "${_iterator}"
     rvm_debug "__list_remote_all${_iterator:-} $rvm_remote_server_url $rvm_remote_server_path"
+
     __list_remote_for "${rvm_remote_server_url}" "$rvm_remote_server_path"
     : $(( _iterator+=1 ))
   done | \command \sort -u | __rvm_version_sort

